deployutils/apps/django/middleware.py

```diff
@@ -49,8 +49,23 @@
 class SessionMiddleware(BaseMiddleware):
 
     def process_request(self, request):
+        JWT_SESSION_STORE = \
+            'deployutils.apps.django.backends.jwt_session_store'
+        JWT_HEADER_NAME = 'HTTP_AUTHORIZATION'
+        JWT_SCHEME = 'bearer'
+
         engine = import_module(django_settings.SESSION_ENGINE)
-        session_key = request.COOKIES.get(settings.SESSION_COOKIE_NAME, None)
+        session_key = None
+        if django_settings.SESSION_ENGINE == JWT_SESSION_STORE:
```

Comment:
Here I am inclined to have two separate `Middleware` classes instead of an if statement `django_settings.SESSION_ENGINE == JWT_SESSION_STORE`.
Alternatively, the `session_key` for  'deployutils.apps.django.backends.encrypted_cookies' should also be passed through the `Authorization` header and both `deployutils.apps.django.backends.encrypted_cookies` and `JWT_SESSION_STORE` tried in sequence until one succeeds.
So maybe what we really need is a `CookieBasedSessionMiddleware` and `AuthHeaderBasedSessionMiddleware`.